.github/workflows/kitctl-test.yml

```diff
@@ -0,0 +1,216 @@
+          for version in "${allVersions[@]}"; do
+            K8S_VERSION=$version
+            version=${version//./-}
+            GUEST_CLUSTER_NAME="guest-${os_type}-${version}-${{ github.run_id }}-${{github.run_attempt}}"
+            cat <<-EOF | kubectl apply -f -
+              apiVersion: kit.k8s.sh/v1alpha1
+              kind: ControlPlane
+              metadata:
+                name: ${GUEST_CLUSTER_NAME} # Desired Cluster name
+              spec:
+                kubernetesVersion: "${K8S_VERSION}"
+                etcd:
+                  replicas: 3
+                master:
+                  apiServer:
+                    replicas: 1  
+          EOF
+            for (( time=0; time<=600; time+=5 )); do
+              controlplane_object=$(kubectl get controlplane ${GUEST_CLUSTER_NAME}  -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+              if [[ "$controlplane_object" == "True" ]]; then
+                echo "Control plane objects are ready"
+                break;
+              else
+                echo "Control plane objects are not ready"
+              fi
+              if [[ "$time" -eq 590 ]]; then
+                echo "Timeout: Control plane objects are not ready after 10 minutes"
+                kubectl logs deploy/kit-controller -n kit
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl wait --for=condition=Ready pods  --all --timeout=5m
+            GUESTKUBECONFIG=/tmp/kubeconfig
+            kubectl get secret ${GUEST_CLUSTER_NAME}-kube-admin-config -ojsonpath='{.data.config}' | base64 -d >$GUESTKUBECONFIG
+            cat <<-EOF | kubectl apply -f -
+              apiVersion: kit.k8s.sh/v1alpha1
+              kind: DataPlane
+              metadata:
+                name: ${GUEST_CLUSTER_NAME}-nodes
+              spec:
+                clusterName: ${GUEST_CLUSTER_NAME} # Associated Cluster name
+                nodeCount: 2
+          EOF
+            for (( time=0; time<=600; time+=5 )); do
+              dataplane_object=$(kubectl get dataplane ${GUEST_CLUSTER_NAME}-nodes  -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+              if [[ "$dataplane_object" == "True" ]]; then
+                echo "Dataplane objects are ready"
+                break;
+              else
+                echo "Dataplane objects are not ready"
+              fi
+              if [[ "$time" -eq 590 ]]; then
+                echo "Timeout: Dataplane objects are not ready after 10 minutes"
+                kubectl logs deploy/kit-controller -n kit 
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl --kubeconfig=$GUESTKUBECONFIG apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/release-1.10/config/master/aws-k8s-cni.yaml
+            kubectl --kubeconfig=$GUESTKUBECONFIG rollout status daemonset/aws-node -n kube-system
```

Comment:
We should specify the --timeout flag in case something wrong happen.
Currently it may stuck forever here until GH action timeout.